core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -624,6 +624,46 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(store.getRemoteBytes("list1").isEmpty)
   }
 
+  Seq(0, Int.MaxValue - 512).foreach { maxRemoteBlockSizeFetchToMem =>
```

Comment:
We could use `Seq(true, false)` instead and name the test as `test for Block replication retry logic with stream = true/false`. And in the test, we could have a variable, e.g. `maxRemoteBlockSizeFetchToMem`, to set the size(0, Int.MaxValue - 512). Thus, the test name could be more understanding.